mira/sources/vensim.py

```diff
@@ -0,0 +1,360 @@
+        state_arg_sympy = process_expression_text(state_arg_text, symbols)
+
+        # map of states to rate laws that affect the state
+        state_sympy_map[state_name] = state_arg_sympy
+
+        # Create a map of states and whether the rate-law/s involved with a state are going in
+        # or out of the state
+        if state_arg_sympy.args:
+            # if it's just the negation of a single symbol
+            if (
+                sympy.core.numbers.NegativeOne() in state_arg_sympy.args
+                and len(state_arg_sympy.args) == 2
+            ):
+                str_symbol = str(state_arg_sympy)
+                state_rate_map[state_name]["outputs"].append(str_symbol[1:])
+            else:
+                for rate_free_symbol in state_arg_sympy.args:
+                    str_rate_free_symbol = str(rate_free_symbol)
+                    if "-" in str_rate_free_symbol:
+                        # Add the symbol to outputs symbol without the negative sign
+                        state_rate_map[state_name]["outputs"].append(
+                            str_rate_free_symbol[1:]
+                        )
+                    else:
+                        state_rate_map[state_name]["inputs"].append(
+                            str_rate_free_symbol
+                        )
+        else:
+            # if it's just a single symbol (i.e. no negation), args property will be empty
+            state_rate_map[state_name]["inputs"].append(str(state_arg_sympy))
+
+    # process initials, just append 0 to each state to represent state at timestamp 0
+    mira_initials = {}
+    for state_name, state_concept in concepts.items():
+        initial = Initial(
+            concept=concepts[state_name].copy(deep=True),
+            expression=safe_parse_expr(state_name + "0"),
+        )
+        mira_initials[initial.concept.name] = initial
+
+    # process parameters
+    mira_parameters = {}
+    for name, expression in new_var_expression_map.items():
+        if expression.replace(".", "").replace(" ", "").isdecimal():
+            model_parameter_info = model_doc_df[model_doc_df["Py Name"] == name]
+            if model_parameter_info["Units"].values[0]:
+                unit_text = (
+                    model_parameter_info["Units"].values[0].replace(" ", "")
+                )
+                parameter = {
+                    "id": name,
+                    "value": float(expression),
+                    "description": model_parameter_info["Comment"].values[0],
+                    "units": {"expression": unit_text},
+                }
+            else:
+                parameter = {
+                    "id": name,
+                    "value": float(expression),
+                    "description": model_parameter_info["Comment"].values[0],
+                }
+            mira_parameters[name] = parameter_to_mira(parameter)
+
+    # add initials as parameters
```

Comment:
This part can be skipped unless there are actual parameters in the input model representing initials